tasks/parallel_verification.py

```diff
@@ -0,0 +1,183 @@
+                    report_code=report_code,
+                    parallel_paths=parallel_paths,
+                ),
+            )
+        if not fas_regular:
+            log.info(
+                "Files and sessions did not match parallel results",
+                extra=dict(
+                    repoid=repoid,
+                    commitid=commitid,
+                    commit_yaml=commit_yaml,
+                    report_code=report_code,
+                    parallel_paths=parallel_paths,
+                ),
+            )
+        if not chunks_legacy:
+            log.info(
+                "Legacy chunks did not match parallel results",
+                extra=dict(
+                    repoid=repoid,
+                    commitid=commitid,
+                    commit_yaml=commit_yaml,
+                    report_code=report_code,
+                    parallel_paths=parallel_paths,
+                ),
+            )
+        if not chunks_regular:
+            log.info(
+                "Chunks did not match parallel results",
+                extra=dict(
+                    repoid=repoid,
+                    commitid=commitid,
+                    commit_yaml=commit_yaml,
+                    report_code=report_code,
+                    parallel_paths=parallel_paths,
+                ),
+            )
+
+        verification_result = (
+            (1 if fas_legacy else 0)
+            + (1 if fas_regular else 0)
+            + (1 if chunks_legacy else 0)
+            + (1 if chunks_regular else 0)
+        ) / 4
+
+        if verification_result == 1:
+            log.info(
+                "Parallel upload processsing verification succeeded",
+                extra=dict(
+                    repoid=repoid,
+                    commitid=commitid,
+                    commit_yaml=commit_yaml,
+                    report_code=report_code,
+                    parallel_paths=parallel_paths,
+                ),
+            )
+        else:
+            log.info(
+                f"Parallel upload processsing verification failed with {verification_result}",
+                extra=dict(
+                    repoid=repoid,
+                    commitid=commitid,
+                    commit_yaml=commit_yaml,
+                    report_code=report_code,
+                    parallel_paths=parallel_paths,
+                ),
+            )
+
+        return
+
+
+# To filter out values not relevant for verifying report content correctness. We
+# don't want these values to be the reason why there exists a diff, since it's
+# not actually related to coverage report contents
+def sort_and_stringify_report_json(d):
+    for session_id in d["sessions"].keys():
+        d["sessions"][session_id].pop("d", None)  # remove timestamp
+        d["sessions"][session_id].pop("e", None)  # remove env
+        d["sessions"][session_id].pop("p", None)  # remove state
+        d["sessions"][session_id].pop("se", None)  # remove session extras
```

Comment:
python passes function arguments by reference so this will actually modify the passed-in `d` variable, and if `d` is or contains a reference to a db model field, you could accidentally overwrite real data by doing this

your parallel and non-legacy cases are safe, but your legacy case could potentially remove these fields from the saved json of a real session. not sure - maybe django returns copies or the `ArchiveField` thing saves us or we'd have to explicitly tell django to save or something

to be safe, probably add `import copy` and add `d = copy.deepcopy(d)` before the for loop :v